lib/src/HttpResponseImpl.cc

```diff
@@ -188,6 +195,56 @@ HttpResponsePtr HttpResponse::newHttpViewResponse(const std::string &viewName,
     return genHttpResponse(viewName, data);
 }
 
+HttpResponsePtr HttpResponse::newFileResponse(
+    unsigned char *pBuffer,
+    const std::string &attachmentFileName,
+    ContentType type,
+    int bufferLength)
+{   
+    // Make Raw HttpResponse
+    auto resp = std::make_shared<HttpResponseImpl>();
+
+    // Convert to sstream
+    std::ostringstream os;
+    for(int i = 0; i < bufferLength; i++)
+    {
+        os << pBuffer[i];
+    }
+
+    // Set response body and length
+    resp->setBody(std::move(os.str()));
+
```

Comment:
resp->setBody(std::string(pBuffer,bufferLenth));

Reply:
This doesn't work, so I am still using the current approach

Reply:
I don't understand why it doesn't work to make a string by pBuffer and bufferLength and set it to the body.
The method of using ostringstream is very inefficient.

Reply:
It was due to a type mistmatch, the incoming buffer is unsigned while the resp body requires a signed format. I fixed this with a static re-cast that you can find in my lastest commit